test_utils/sema_utils/utils.go

```diff
@@ -40,8 +40,9 @@ func ParseAndCheck(t testing.TB, code string) (*sema.Checker, error) {
 }
 
 type ParseAndCheckOptions struct {
+	MemoryGauge      common.MemoryGauge
```

Comment:
One of the main improvements: Allow specifying the memory gauge for the parser and checker in the options, instead of as a function parameter. This reduces the combinatorial explosion of test utils.

Reply:
Good idea!